akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraOffsetStoreImpl.scala

```diff
@@ -4,36 +4,23 @@
 
 package akka.projection.cassandra.internal
 
-import java.util.UUID
-
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
 
 import akka.Done
 import akka.annotation.InternalApi
-import akka.persistence.query
 import akka.projection.ProjectionId
 import akka.projection.cassandra.CassandraOffsetStore
+import akka.projection.internal.OffsetSerialization
 import akka.stream.alpakka.cassandra.scaladsl.CassandraSession
-import com.datastax.oss.driver.api.core.cql.Row
-
-/**
- * INTERNAL API
- */
-@InternalApi private[akka] object CassandraOffsetStoreImpl {
-  private val StringManifest = "STR"
-  private val LongManifest = "LNG"
-  private val IntManifest = "INT"
-  private val SequenceManifest = "SEQ"
-  private val TimeBasedUUIDManifest = "TBU"
-}
 
 /**
  * INTERNAL API
  */
 @InternalApi private[akka] class CassandraOffsetStoreImpl(session: CassandraSession)(implicit ec: ExecutionContext)
     extends CassandraOffsetStore {
-  import CassandraOffsetStoreImpl._
+  import OffsetSerialization.fromStorageRepresentation
+  import OffsetSerialization.toStorageRepresentation
```

Comment:
@patriknw, can we give this another iteration? I'd rather allow users to inject two functions instead of hardcoding `OffsetSerialization._`.

I'm thinking about a projection where the source is Kafka but I'm using Cassandra's `OffsetStore` anyway. I'm using Alpakka Kafka so in my projection the `Offset` type is none of the above but a [`PartitionOffset`](https://doc.akka.io/api/alpakka-kafka/2.0.2/akka/kafka/ConsumerMessage$$PartitionOffset.html).

I think `akka-projection` should include this `OffsetSerialization` as an example/default pair of functions to read/write the most common offset formats, but I think it'd be better if each projection implementation left to the user to provide/override those two functions.

So, in [tests](https://github.com/akka/akka-projection/blob/4544eeaaf2200c3e97d35512e12f53c9ee73553b/akka-projection-cassandra/src/test/scala/akka/projection/cassandra/CassandraProjectionSpec.scala#L143-L147) things align and work nicely because tests use `Long` as the `Offset`:

https://github.com/akka/akka-projection/blob/4544eeaaf2200c3e97d35512e12f53c9ee73553b/akka-projection-cassandra/src/test/scala/akka/projection/cassandra/CassandraProjectionSpec.scala#L32-L47

I was thinking whether this Offset de/ser mechanism I'm thinking of should use the Akka serialization extension but I read in the discussion in this PR it's a requirement to store the offsets in human-readable format (which makes perfect sense to me).

Reply:
We will have support for Kafka offsets out of the box. For discussion of String vs binary, let's  have that in https://github.com/akka/akka-projection/issues/60

Also remember that we are doing MVP now, so custom offset types might be out of scope.

Reply:
I missed #60. LGTM. :-)